src/components/posts/PostsPage.js

```diff
@@ -1,71 +1,154 @@
-import React from 'react';
-import { useState, useEffect } from 'react';
-import PostForm from './PostForm';
-import client from '../../utils/client';
-import './style.css';
-import { Box, Stack } from '@mui/material';
-import Header from '../Header/Header';
-import dateTimetoRelativeTime from './helperfunctions';
-import { Link } from 'react-router-dom';
+import React from 'react'
+import { useState, useEffect } from 'react'
+import PostForm from './PostForm'
+import client from '../../utils/client'
+import './style.css'
+import { Box, Stack } from '@mui/material'
+import Header from '../Header/Header'
+import dateTimetoRelativeTime from './helperfunctions'
+import { Link } from 'react-router-dom'
 
 const PostsPage = ({ role }) => {
-  const [post, setPost] = useState({ content: '' });
-  const [postResponse, setPostResponse] = useState('');
-  const [posts, setPosts] = useState([]);
+  const [post, setPost] = useState({ content: '' })
+  const [postResponse, setPostResponse] = useState('')
+  const [posts, setPosts] = useState([])
+  const [comment, setComment] = useState('')
+  const [load, setLoad] = useState(true)
+  const [showAllArr, setShowAll] = useState([])
 
   useEffect(() => {
-    client.get('/posts').then((res) => setPosts(res.data.data.posts));
-  }, []);
-
+    client.get('/posts').then((res) => {
+      setPosts(res.data.data.posts)
+    })
+  }, [load])
   const createPost = async (event) => {
-    event.preventDefault();
+    event.preventDefault()
     client
       .post('/post', post)
       .then((res) => {
-        setPostResponse(res.data);
-        setPosts((posts) => [res.data.data.post, ...posts]);
+        setPostResponse(res.data)
+        setPosts((posts) => [res.data.data.post, ...posts])
+        setLoad((x) => !x)
+      })
+      .catch((data) => {
+        console.log(data)
       })
-      .catch((data) => {});
-    setPost(() => ({ content: '' }));
-  };
+    setPost(() => ({ content: '' }))
+  }
 
   const handleChange = (event) => {
-    event.preventDefault();
-    const { value, name } = event.target;
+    event.preventDefault()
+    const { value, name } = event.target
     setPost({
       ...post,
       [name]: value,
-    });
-  };
+    })
+  }
+
+  const createComment = async (event, postId) => {
```

Comment:
I don't think you need to use `async` here as you are not making an `await` call in the method?

Reply:
The codebase was like that from the beginning and we forgot to change it. will change it!